avocado/core/runners/sysinfo.py

```diff
@@ -0,0 +1,188 @@
+
+    def _save_sysinfo(self, log_hook):
+        try:
+            file_path = os.path.join(self.sysinfo_dir, log_hook.name)
+            for data in log_hook.collect():
+                self.queue.put(messages.FileMessage.get(data, file_path))
+        except sysinfo_collectible.CollectibleException as e:
+            self.queue.put(messages.LogMessage.get(e.args[0]))
+        except Exception as exc:  # pylint: disable=W0703
+            self.queue.put(messages.StderrMessage.get("Collection %s failed: %s"
+                                                      % (type(log_hook), exc)))
+
+    def collect(self):
+        """Log all collectibles at the start of the event."""
+        self._set_collectibles()
+        for log_hook in self.collectibles:
+            self._save_sysinfo(log_hook)
+
+        if self.log_packages:
+            self._log_packages(self.sysinfo_dir)
+        self.queue.put(messages.FinishedMessage.get('pass'))
+
+    def _log_packages(self, path):
+        installed_path = os.path.join(path, "installed_packages")
+        installed_packages = "\n".join(self._get_installed_packages()) + "\n"
+        self.queue.put(messages.FileMessage.get(installed_packages,
+                                                installed_path))
+
+
+class PostSysInfo(PreSysInfo):
+    """
+    Log different system properties after end event.
+
+    An event may be a job, a test, or any other event with a
+    beginning and end.
+    """
+
+    sysinfo_dir = os.path.join('sysinfo', 'post')
+
+    def __init__(self, config, sysinfo_config, queue, test_fail=False):
+        """
+        :param test_fail: flag for fail tests. Default False
+        :type test_fail: bool
+        """
+        self.test_fail = test_fail
+        super().__init__(config, sysinfo_config, queue)
+
+    def _set_collectibles(self):
+        super()._set_collectibles()
+        if self.test_fail:
+            for fail_cmd in self.sysinfo_config.get("fail_commands", []):
+                self.collectibles.add(
+                    sysinfo_collectible.Command(fail_cmd, timeout=self.timeout,
+                                                locale=self.locale))
+
+            for fail_filename in self.sysinfo_config.get("fail_files", []):
+                self.collectibles.add(sysinfo_collectible.Logfile(fail_filename))
+
+
+class SysinfoRunner(nrunner.BaseRunner):
+    """
+    Runner for gathering sysinfo
+
+    Runnable attributes usage:
+
+     * uri: sysinfo type pre/post. This variable decides if the sysinfo is
+            collected before or after the test.
+
+     * kwargs: "sysinfo" dictionary with commands/tasks which should be
+               performed during the sysinfo collection.
+    """
+
+    def run(self):
+        yield self.prepare_status('started')
+        sysinfo_config = self.runnable.kwargs.get('sysinfo', {})
+        test_fail = self.runnable.kwargs.get('test_fail', False)
+        if self.runnable.uri in ['pre', 'post']:
```

Comment:
One suggestion, if you change the logic for `if self.runnable.uri not in ['pre', 'post']:` and fail, you don't need the else and you can reduce one indentation for the code that does the job.